### PR TITLE
Fix drop of OVAL checks extending non-existing definitions

### PR DESCRIPTION
**Pull request by yuumasato**
Do not remove checks while iterating over them

With inbuilt python element tree it is not safe to remove itens from the
list you are iterating on.

### CODE REVIEW

**Comment by yuumasato** [comment]
This will fix build of content with OVAL 5.10 only.

**Comment by mpreisler** [comment]
Good catch, this is pretty much a no brainer.

File: shared/utils/relabel-ids.py [hunk 1]
```diff
@@ -216,11 +217,13 @@ def drop_oval_checks_extending_non_existing_checks(ovaltree):
...
+                defstoremove.append(definition)
 
-                sys.stderr.write("WARNING: OVAL check '%s' extends non-existing '%s' "
-                                 "OVAL definition, removing check from OVAL definitions.\n"
-                                 % (definition.get("id"), extdefinitionref))
-                definitions.remove(definition)
+    for definition in defstoremove:
+        sys.stderr.write("WARNING: OVAL check '%s' extends non-existing '%s' "
+                "OVAL definition, removing check from OVAL definitions.\n"
+                % (definition.get("id"), extdefinitionref))
```
Comment (mpreisler) [inline shared/utils/relabel-ids.py]: extdefinitionref will be wrong in this context

**Comment by mpreisler** [comment]
Hmm just as I said it I found a major issue, the variable used when logging the message will be set to the wrong value.

File: shared/utils/relabel-ids.py (hunk 1 above)
Comment (yuumasato) [inline shared/utils/relabel-ids.py]: Ouch.

Comment (yuumasato) [inline shared/utils/relabel-ids.py]: Fixed,  :)

**Comment by mpreisler** [comment]
ACK

**Pull request merged**